### PR TITLE
Perpetuance job ability implementation

### PR DESCRIPTION
**Pull request by nfaletra**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Logic for perpetuance increasing enhancing duration was implemented but the status effect and removal logic was missing. Added the ability to the db with the correct light arts flags and removal of the status effect on the next white magic enhancing spell.

Also fixed light arts status effect flags as it was not properly being removed when logging out. I double checked that behavior on retail to make sure I wasn't crazy.

## Steps to test these changes

* Cast enhancing magic on yourself (e.g. Regen)
* Use Lights Arts -> Perpetuance
* Cast same enhancing magic on yourself
* Observe the duration is doubled compared to the first cast

### CODE REVIEW

File: scripts/globals/abilities/perpetuance.lua [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    return 0, 0
+end
+
+abilityObject.onUseAbility = function(player, target, ability)
+    player:addStatusEffect(xi.effect.PERPETUANCE, 1, 0, 60)
+
+    return xi.effect.PERPETUANCE
+end
+
+return abilityObject
```
Comment (Xaver-DaRed) [inline scripts/globals/abilities/perpetuance.lua]: Need a newline at the end of the file, for CI.

File: src/map/utils/charutils.cpp [hunk 2]
```diff
@@ -5912,6 +5912,10 @@ namespace charutils
             {
                 PChar->StatusEffectContainer->DelStatusEffect(EFFECT_ACCESSION);
             }
+            if (PSpell->getSkillType() == SKILL_ENHANCING_MAGIC)
+            {
```
Comment (Xaver-DaRed) [inline src/map/utils/charutils.cpp:5916]: Correct me if I'm wrong, but wouldnt this need a check for the spell group, so only white magic enhancing skills (AKA, not spikes) count?

Comment (nfaletra) [reply src/map/utils/charutils.cpp:5916]: it's within a spell group check.

File: scripts/globals/abilities/perpetuance.lua (hunk 1 above)
Comment (nfaletra) [reply scripts/globals/abilities/perpetuance.lua]: Done and done.

File: src/map/utils/charutils.cpp (hunk 2 above)
Comment (Xaver-DaRed) [reply src/map/utils/charutils.cpp:5916]: Very true, I saw it. I stand corrected

**Review by Xaver-DaRed — APPROVED** [review]

**Review by claywar — APPROVED** [review]

**Pull request merged**